### DISCUSSION TITLE
Strip quotes in yaml for strings.

### DISCUSSION THREAD

**Discussion by warmfire540** [prettier/prettier#16956] [Q&A]
I'm trying to remove the quotes from strings in yaml.  This doesn't seem possible with default settings so I figured I'd extend it w/ a plugin.

so
```yaml
foo: "bar baz"
```

will become
```yaml
foo: bar baz
```

I am doing the `print` function, and it works if I make the output as expected, however as soon as I change something (remove quotes) - it yells..  Any help?

Here's my plugin w/ some attempts left in place

```javascript
const { parsers, printers } = require("prettier/parser-yaml");
const { doc } = require("prettier");
const {concat,  group, indent, join, line, softline } = doc.builders;

const isValidWithoutQuotes = (str) => {
    if (/^[0-9]/.test(str)) return false; // Starts with a number
    if (/[\s]|[:{}[\],&*#?|-]|^(yes|no|true|false|null|~)$/.test(str)) return false; // Special characters or reserved words
    return true;
};

const quoteRemoverPlugin = {
    parsers: {  
        yaml: {
            ...parsers.yaml,
        }
    },
    printers: {
        yaml: {
            ...printers.yaml,
            print(path, options, print) {
                const { node } = path;
                const parts = [];

                if (
                    node &&
                    ["quoteDouble" ,"quoteSingle"].includes(node.type) &&
                    node.value === 'bar' &&
                    isValidWithoutQuotes(node.value)
                ) {
                    // Modify the node to remove quotes
                    //node.type = "quoteDouble";
                    //console.log('node', node.type, node.value, options);
                    //return print("wee");
                    //console.log('node', node.type, node.value, node);
                    // return printers.yaml.print({
                    //     ...path,
                    //     node: {
                    //         ...path.node,
                    //         type: "plain"
                    //     }
                    // }, options, print);
                    //node.type = "plain";
                    //options.singleQuote = true;
                    //path.node.type = "plain";
                    //node.type = "plain";
                    parts.push("");
                    parts.push(node.value);
                    //parts.push(softline);
                    parts.push("");
                    console.log(node)
                    return group(parts);
                }
                
                return printers.yaml.print(path, options, print);
            },
        }
    }
};

module.exports = quoteRemoverPlugin;
```

It works if my plugin doesn't change anything.. here is output from `--debug-check`
<img width="1000" alt="image" src="https://github.com/user-attachments/assets/1e978504-4305-4095-ac3d-dc7cf9896fa6" />

here is the output if I "comment out" adding the quotes, it complains the quotes are missing I guess?
<img width="976" alt="image" src="https://github.com/user-attachments/assets/bcab9831-eb01-48e5-b024-ee686d0ad7aa" />

**Comment by fisker** [comment]
Since your print logic changed the AST, to make `ast(input) === ast(prettier(input))` pass, you need fix the AST in the [`clean`](https://github.com/prettier/prettier/blob/d9d0999aac70c04b06b35cbcb2c804b4a9c8d8a8/src/language-yaml/printer-yaml.js#L405) function.

**Comment by warmfire540** [comment]
> you need fix the AST in the [clean](https://github.com/prettier/prettier/blob/d9d0999aac70c04b06b35cbcb2c804b4a9c8d8a8/src/language-yaml/printer-yaml.js#L405) function.

how do I do that as a plugin w/ the available APIs?  I see it's passed as `massageAstNode` - am I overriding this func?
